src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private T [] array = (T[]) new  Object [10];
+    private int pointer;
+
     @Override
     public void add(T value) {
-
+        if (pointer == array.length) {
+            grow(array);
+        }
+        array[pointer++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > pointer) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Test failed! Can't correctly add element by index ");
+        }
+        if (index <= pointer) {
+            T[] tmp = (T[]) new Object[array.length - index];
```

Comment:
Instead of manually copying array elements, use `System.arraycopy()` for better performance and readability.